packages/devtools_app/lib/src/service_manager.dart

```diff
@@ -764,14 +779,16 @@ class IsolateManager extends Disposer {
 
 /// Manager that handles tracking the service extension for the main isolate.
 class ServiceExtensionManager extends Disposer {
-  ServiceExtensionManager(this._mainIsolate);
+  ServiceExtensionManager(this._mainIsolate, this._getIsolateCached);
 
   VmServiceWrapper _service;
 
   bool _checkForFirstFrameStarted = false;
 
   final ValueListenable<IsolateRef> _mainIsolate;
 
+  final IsolateF _getIsolateCached;
```

Comment:
why not just use the serviceManager isolate cache directly instead of passing in this function?

Reply:
Now using the pre existing cache :)